src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -1447,7 +1447,7 @@ void UpdateMemoryCost(in UInt256 position, in UInt256 length)
 
                         ZeroPaddedSpan returnDataSlice = _returnDataBuffer.SliceWithZeroPadding(src, (int) length);
                         vmState.Memory.Save(in dest, returnDataSlice);
-                        if (_txTracer.IsTracingInstructions) _txTracer.ReportMemoryChange((long) dest, returnDataSlice.ToArray());
+                        if (_txTracer.IsTracingInstructions) _txTracer.ReportMemoryChange((long) dest, returnDataSlice);
```

Comment:
Passing access to EVM buffer to an externally implemented interface.

Reply:
same as above